### PR TITLE
Switch to RSpec

### LINKED ISSUE #22: Switch from MiniTest to RSpec
I've already felt the loss of `describe`/`context` blocks when writing tests in #13 and #18. @rewinfrey and I discussed Rails testing frameworks and would love to switch to [RSpec](https://github.com/rspec/rspec-rails).

### PR DESCRIPTION
**Pull request by rewinfrey**
Closes #22 

❤️ for the RSpec! 

/cc @cheshire137

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#
+# Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
+
+# Checks for pending migration and applies them before tests are run.
+# If you are not using ActiveRecord, you can remove this line.
+ActiveRecord::Migration.maintain_test_schema!
+
+RSpec.configure do |config|
+  # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
```
Comment (cheshire137) [inline spec/rails_helper.rb]: I think we could drop this. I add FactoryGirl for factories in one of my branches, though I don't know if it's merged yet.

**Comment by rewinfrey** [comment]
@cheshire137 I transferred your model tests from `test/models` to `spec/models` from https://github.com/cheshire137/overwatch-team-comps/pull/18 . Are you cool with merging this as is? I also removed the `fixture` config in `rails_helper` as you suggested 😄

**Comment by cheshire137** [comment]
Thank you for translating my MiniTest tests to RSpec! Yep, this looks good to me. 👌

**Comment by cheshire137** [comment]
This also now fixes #6 since we merged in #18.

**Pull request merged**